cmd/difference.go

```diff
@@ -53,17 +54,29 @@ func (d differType) String() string {
 	return "unknown"
 }
 
-func objectDifference(sourceClnt, targetClnt Client, sourceURL, targetURL string) (diffCh chan diffMessage) {
-	return difference(sourceClnt, targetClnt, sourceURL, targetURL, true, false, DirNone)
+func objectDifference(sourceClnt, targetClnt Client, sourceURL, targetURL string, excludeOptions []string) (diffCh chan diffMessage) {
+	return difference(sourceClnt, targetClnt, sourceURL, targetURL, true, false, DirNone, excludeOptions)
 }
 
 func dirDifference(sourceClnt, targetClnt Client, sourceURL, targetURL string) (diffCh chan diffMessage) {
-	return difference(sourceClnt, targetClnt, sourceURL, targetURL, false, true, DirFirst)
+	return difference(sourceClnt, targetClnt, sourceURL, targetURL, false, true, DirFirst, nil)
+}
+
+func matchExcludeOptions(excludeOptions []string, srcSuffix string) bool {
+	if excludeOptions != nil {
+		for _, pattern := range excludeOptions {
+			r, err := filepath.Match(pattern, srcSuffix)
```

Comment:
We should add a testCase for this , also perhaps we should use wildcard package from server used by find.go instead than filepath.Match()

An example case is here - https://play.golang.org/p/Ega9qgD4Qz - Since we deal with flat-namespace for both filesystem and object storage we should use this instead.

Reply:
done